src/lib.rs

```diff
@@ -259,3 +271,38 @@ pub fn make_game() -> Result<Game, JsValue> {
         players,
     })
 }
+
+pub fn start_websocket() -> Result<(), JsValue> {
+    let ws = WebSocket::new("ws://localhost:3012")?;
+    let onmessage_callback = Closure::wrap(Box::new(move |e: MessageEvent| {
```

Comment:
`Closure::wrap`
`Box::new`
`move | ... |`

.... so many questions, I barely know where to begin.

Reply:
What I know:

boxing something allocates it on the heap, a useful place for long lived data

Closure::wrap is a thing for converting a rust function into a JS function

`|| {}` is the syntax for an inline function. `move || {}` something something borrow checker move references _aimless arm motions_